### PR TITLE
Atanas/grok 12797 bio demo dendrogram with sequence space

### PR DESCRIPTION
**Pull request by tanas80**
Bio demos: Similarity, Diversity, Sequence Space, Activity Cliffs

### CODE REVIEW

File: packages/Bio/src/demo/bio01a-hierarchical-clustering-and-sequence-space.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    await step('Building sequence space.', async () => {
+      spViewer = await demoSequenceSpace(view, df, method);
+    })();
+
+    await step('Hierarchical clustering.', async () => {
+      const seqCol: DG.Column<string> = df.getCol('sequence');
+      const seqList = seqCol.toList();
+      const distance: DistanceMatrix = DistanceMatrix.calc(seqList, (aSeq: string, bSeq: string) => {
+        const levDistance = lev.distance(aSeq, bSeq);
+        return levDistance / ((aSeq.length + bSeq.length) / 2);
```
Comment (LaykDimon) [inline packages/Bio/src/demo/bio01a-hierarchical-clustering-and-sequence-space.ts:50]: what is this `levDistance / ((aSeq.length + bSeq.length) / 2);`?
maybe put it into const and then return the const to be more clear?

File: packages/Bio/src/demo/bio01b-hierarchical-clustering-and-activity-cliffs.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        .find('button.scatter_plot_link,cliffs_grid').get()[0] as HTMLButtonElement;
+      cliffsLink.click();
+    })();
+
+    await step('Hierarchical clustering.', async () => {
+      const seqCol: DG.Column<string> = df.getCol('sequence');
+      const seqList = seqCol.toList();
+      const distance: DistanceMatrix = DistanceMatrix.calc(seqList, (aSeq: string, bSeq: string) => {
+        const levDistance = lev.distance(aSeq, bSeq);
+        return levDistance / ((aSeq.length + bSeq.length) / 2);
```
Comment (LaykDimon) [inline packages/Bio/src/demo/bio01b-hierarchical-clustering-and-activity-cliffs.ts:65]: same as mentioned above

File: packages/Bio/src/demo/utils.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+enum EMBED_COL_NAMES {
+  X = 'Embed_X',
+  Y = 'Embed_Y'
+}
+
+export async function demoSequenceSpace(
+  view: DG.TableView, df: DG.DataFrame, method: string
+): Promise<DG.ScatterPlotViewer> {
+  let resSpaceViewer: DG.ScatterPlotViewer;
+  if (true) {
```
Comment (LaykDimon) [inline packages/Bio/src/demo/utils.ts:32]: this is not the best approach
what is the condition here?

File: packages/Bio/src/demo/bio01a-hierarchical-clustering-and-sequence-space.ts (hunk 1 above)
Comment (tanas80) [reply packages/Bio/src/demo/bio01a-hierarchical-clustering-and-sequence-space.ts:50]: Typization

File: packages/Bio/src/demo/bio01b-hierarchical-clustering-and-activity-cliffs.ts (hunk 2 above)
Comment (tanas80) [reply packages/Bio/src/demo/bio01b-hierarchical-clustering-and-activity-cliffs.ts:65]: the same

File: packages/Bio/src/demo/utils.ts (hunk 3 above)
Comment (tanas80) [reply packages/Bio/src/demo/utils.ts:32]: comment to clarify switching between two implementations of sequence space

**Pull request merged**